games/Space_Invasion.js

```diff
@@ -0,0 +1,572 @@
+663L606.6666666.
+66666...........
+66..............`],
+)
+
+const space=[
+  map`
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................
+.................`
+]
+
+const start = tune`
+250: f5^250 + f4^250,
+250: d4^250 + g5^250,
+250: e4^250 + a5^250,
+250: f4^250,
+250: d4^250 + f5^250,
+250: e4^250 + g5^250,
+250: f4^250 + a5^250,
+250: d4^250,
+250: e4^250 + d5^250,
+250: f4^250 + e5^250,
+250: d4^250 + f5^250,
+250: e4^250,
+250: f4^250,
+250: d4^250 + e5^250,
+250: e4^250 + f5^250,
+250: f4^250 + g5^250,
+250: d4^250,
+250: e4^250,
+250: f4^250 + g5^250,
+250: e4^250 + a5^250,
+250: f4^250 + b5^250,
+250: g4^250,
+250: e4^250,
+250: f4^250,
+250: g4^250,
+250: g5^250,
+250: f4^250 + a5^250,
+250: g4^250 + b5^250,
+250: a4^250,
+250: f4^250,
+250: g4^250 + a5^250,
+250: a4^250 + b5^250`
+const boom = tune`
+161.29032258064515,
+161.29032258064515: c4/161.29032258064515 + d4~161.29032258064515 + e4^161.29032258064515,
+4838.709677419354`
+const death= tune`
+1200,
+150: e5-150 + d5-150,
+150: e5^150 + d5-150,
+150: d5-150 + e5^150 + c5/150,
+150: c5-150 + d5-150 + b4^150,
+150: c5-150 + b4-150 + a4^150 + d5/150,
+150: b4-150 + a4^150,
+150: a4-150 + g4^150 + b4-150,
+150: g4-150 + f4^150 + e4^150 + a4-150,
+150: e4^150 + d4^150 + c4^150 + g4-150 + f4-150,
+150: f4-150 + e4-150 + d4~150 + c4/150,
+150: e4-150 + d4-150 + c4-150,
+150: c4-150,
+1800`
+
+setMap(space[0])
+setBackground(bg)
+
+const clear_screen = () =>{
+  if(start_check == 0){
+    let x = 0
+    clearInterval(start_screen)
+    addSprite(8, 15, ship)
+    
+    clearText()
+    start_check = 1
+  }
+}
+
+const update = () =>{
+  const HUD_update = () =>{
+    clearText()
+    addText("Score:"+`${score}`,{ 
+      x: 2, 
+      y: 0, 
+      color: [ 255, 255, 255] 
+    })
+    addText("HP:",{ 
+      x: 12, 
+      y: 0, 
+      color: [ 255, 255, 255] 
+    })
+/*    addText("Pow:",{ 
+      x: 11, 
+      y: 1, 
+      color: [ 255, 255, 255] 
+    })
+*/  
+    if(lives == 3){
+      clearTile(14,0)
+      clearTile(15,0)
+      clearTile(16,0)
+      addSprite(14,0,heart)
+      addSprite(15,0,heart)
+      addSprite(16,0,heart)
+    }
+    else if(lives == 2){
+      clearTile(14,0)
+      clearTile(15,0)
+      clearTile(16,0)
+      addSprite(14,0,heart)
+      addSprite(15,0,heart)
+      addSprite(16,0,border)
+    }
+    else if(lives == 1){
+      clearTile(14,0)
+      clearTile(15,0)
+      clearTile(16,0)
+      addSprite(14,0,heart)
+      addSprite(15,0,border)
+      addSprite(16,0,border)
+    }
+    else if(lives == 0){
+      clearTile(14,0)
+      clearTile(15,0)
+      clearTile(16,0)
+      addSprite(14,0,border)
+      addSprite(15,0,border)
+      addSprite(16,0,border)
+    }
+/*
+    if(sbattery == 3){
```

Comment:
yeah,  I've commented some of the function which were not implemented. And I left them behind if anyone else wants to modify the given code.

And I'm not sure why that error arises, but I have few assumptions like it happens with spawn killing of ships, when the laser and just spawned ship have same location.
It might have to do something with getFirst function....if the sprite doesnt exist...it gives the error above.

Reply:
ah! good idea about code for remixes :)

Reply:
when the error appears, the shots end, is it your idea that the shots end?

Reply:
nope, the bullet function gets bricked when that happens

Reply:
strangely the error isn't appearing up for me anymore, is it appearing up for you?

Reply:
because the error doesn't always appear, I'll merge your game :)
If you're a teen, you can get your sprig here! https://sprig-order.hackclub.dev/

Reply:
lemme see, if I can fix that error